src/ui/public/vislib_vis_type/vislib_renderbot.js

```diff
@@ -46,7 +46,10 @@ module.exports = function VislibRenderbotFactory(Private) {
   VislibRenderbot.prototype.buildChartData = buildChartData;
   VislibRenderbot.prototype.render = function (esResponse) {
     this.chartData = this.buildChartData(esResponse);
-    this.vislibVis.render(this.chartData, this.uiState);
+    // to allow legend to render first
+    setTimeout(() => {
```

Comment:
legend is waiting for renderbot.chartData to be available. once thats done it'll render in the next digest cycle so its sufficient for renderbot to just wait here after it makes chartData available